src/login/pam_systemd.c

```diff
@@ -430,6 +434,34 @@ _public_ PAM_EXTERN int pam_sm_open_session(
                            "id=%s object_path=%s runtime_path=%s session_fd=%d seat=%s vtnr=%u original_uid=%u",
                            id, object_path, runtime_path, session_fd, seat, vtnr, original_uid);
 
+        r = asprintf(&environ_file, "%s/systemd/environment", runtime_path);
```

Comment:
No _cleanup_, just `environ_file = strjoina(runtime_path, "/systemd/environment")`.

Hm, is a path in /run the best way to convey that information? Maybe it is... We could do it over dbus, but I don't see an actual advantage to that.